docker-plain/files/chart/values.yaml

```diff
@@ -0,0 +1,128 @@
+replicaCount: 1
+
+imagePullSecrets: []
+nameOverride: ""
+fullnameOverride: ""
+
+# NOTE: By default image Values are injected from CICD Jenkins pipeline, values defined here are used when not being on CICD Jenkins pipeline context.
+# Default values here are provided in case one needs to use the chart without CICD (i.e.: testing the chart).
+image:
+  # registry: "docker.io"
+  # path: "nginxinc"
+  # name: "nginx-unprivileged"
+  # tag: "latest"
+  # see https://kubernetes.io/docs/concepts/containers/images/#image-pull-policy
+  pullPolicy: IfNotPresent
+
+## Prefer using ingress over openshift routes
+ingress:
+  enabled: false
+  className: 'openshift-default'
+  # router: external
+  annotations:
+    ## adjust openshift timeouts (default is 60s)
+    haproxy.router.openshift.io/timeout: 300s
+    haproxy.router.openshift.io/timeout-tunnel: 300s
+    ## e.g. add cert-manager support by annotating the ingress https://cert-manager.io/docs/usage/ingress/
+    ## ask in your company for good defaults
+
+  hosts: [] # When defining a host you must define also a path
+  # - host: yourapp.yourdomain.com
+  #   paths:
+  #     - path: /
+  #       pathType: Prefix
+  tls: [] # If `tls` is left empty then the default OpenShift TLS config will be loaded (i.e.: TLS edge termination with HTTP redirect to HTTPS)
+  # - secretName: chart-example-tls
+  #   hosts:
+  #     - yourapp.yourdomain.com
+
+service:
+  enabled: true
+  port: 8080
+  type: ClusterIP
+
+# There are two types of strategy: `Recreate` and `RollingUpdate`
+# Please refer to https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#strategy
+deploymentStrategy:
+  type: RollingUpdate
+  rollingUpdate:
+    maxUnavailable: 0%
+    maxSurge: 50%
+
+probes:
+  livenessProbe:
+    failureThreshold: 3
+    httpGet:
+      path: "/"
+      port: 8080
+      scheme: HTTP
+    initialDelaySeconds: 5
+    periodSeconds: 10
+    successThreshold: 1
+    timeoutSeconds: 3
+
+  readinessProbe:
+    failureThreshold: 1
+    httpGet:
+      path: "/"
+      port: 8080
+      scheme: HTTP
+    initialDelaySeconds: 3
+    periodSeconds: 5
+    successThreshold: 1
+    timeoutSeconds: 3
+
+serviceAccount:
+  # Specifies whether a service account should be created
+  create: false
+  # Annotations to add to the service account
+  annotations: {}
+  # The name of the service account to use.
+  # If not set and create is true, a name is generated using the fullname template
+  # name: "default"
+
+podAnnotations: {}
+
+podSecurityContext: {}
+  # fsGroup: 2000
+
+securityContext: {}
+  # capabilities:
+  #   drop:
+  #   - ALL
+  # readOnlyRootFilesystem: true
+  # runAsNonRoot: true
+  # runAsUser: 1000
+
+resources:
+  limits:
+    cpu: 100m
+    memory: 128Mi
+  requests:
+    cpu: 50m
+    memory: 64Mi
+
+autoscaling:
+  enabled: false
+  minReplicas: 1
+  maxReplicas: 100
+  targetCPUUtilizationPercentage: 80
+  # targetMemoryUtilizationPercentage: 80
```

Comment:
Enabling autoscaling in `values.yaml` doesn't work yet, without the corresponding HPA resource

If a user sets `enabled` to true, currently the behavior is `replicas` will be unset in `deployment.yaml`, it will always have exactly 1 replica, which is the default.

I suggest we either remove this section in values.yaml, or add HorizontalPodAutoscaler resource to make autoscaling work as expected.

By the way, having 100 as the default value of `maxReplicas` seems a bit too many to me. Maybe something below 10 would be good enough?